Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -82,6 +84,7 @@ private struct HTTPParserState {
             let (index, length) = consumeSlice()
             self.currentStatus = self.cumulationBuffer!.getString(at: index, length: length)!
         case .body:
+            self.currentNameIndex = nil
```

Comment:
Can we ever hit this branch without having gone through `.headerValue`? It seems like should be an assertion instead.

Reply:
yep sounds correct... same for the slice imho